examples/step-87/step-87.cc

```diff
@@ -0,0 +1,1175 @@
+        // consisting of the unit point.
+        std::cout << " - Values at point:" << std::endl;
+
+        {
+          FEValues<dim> fe_values(mapping,
+                                  fe,
+                                  Quadrature<dim>(p_unit),
+                                  update_values);
+          fe_values.reinit(cell);
+
+          std::vector<double> quad_values(1);
+          fe_values.get_function_values(vector, quad_values);
+          const double value_0 = quad_values[0];
+          std::cout << "  - " << value_0 << " (w. FEValues)" << std::endl;
+          values_line.push_back(value_0);
+        }
+
+        // The second approach considers FEPointEvaluation, which directly
+        // takes a list of unit points for the subsequent evaluation.
+        {
+          std::vector<double> cell_vector(fe.n_dofs_per_cell());
+          cell->get_dof_values(vector, cell_vector.begin(), cell_vector.end());
+
+          FEPointEvaluation<1, dim> fe_point(mapping, fe, update_values);
+          fe_point.reinit(cell, ArrayView<const Point<dim>>(p_unit));
+          fe_point.evaluate(cell_vector, EvaluationFlags::values);
+          const auto value_1 = fe_point.get_value(0);
+          std::cout << "  - " << value_1 << " (w. FEPointEvaluation)"
+                    << std::endl;
+        }
+
+        // Finally, in the third approach, the function
+        // VectorTools::point_value() is considered. It performs both
+        // the search of the surrounding cell and the evaluation at the
+        // requested point. However, its application is limited
+        // to a serial run of the code.
+        {
+          const auto value_2 =
+            VectorTools::point_value(dof_handler, vector, p_real);
+          std::cout << "  - " << value_2 << " (w. VectorTools::point_value())"
+                    << std::endl;
+          std::cout << std::endl;
+        }
+      }
+
+    // We output the requested points together with the corresponding
+    // evaluated solution to a CSV file.
+    std::cout << " - writing csv file" << std::endl;
+    print_along_line("example_0_profile.csv", points_line, values_line);
+  }
+
+  // Obviously, the code above cannot work for distributed meshes, since
+  // the search (which might require communication) is called within a for-loop
+  // with loop bounds possibly different on each process. In the following
+  // code examples, we present the usage of arbitrary point evaluation
+  // in a parallel computation.
+  //
+  // @sect3{Mini example 1: Evaluation at given points on a distributed mesh}
+  //
+  // Just like in the introductory example, we evaluate the solution
+  // along a line, however, on a distributed mesh. We again start with
+  // setting up the objects needed for a finite element discretization.
+  void example_1()
+  {
+    constexpr unsigned int dim       = 2;
+    constexpr unsigned int fe_degree = 3;
+
+    AssertDimension(dim, 2);
```

Comment:
I am unsure why you are asserting something that is declared as a constexpr above? Is it just to ensure that the user will not "try" to set the problem to 3D?

Reply:
Our intention was that we did not implement everything in a dimension-independent way (mainly the definition of points). With the assert we wanted to avoid exactly what you described. However, if you think it makes sense, we could skip the assert and extend the definitions such that running the tutorial in 1D to 3D should be possible.

Reply:
I think adding a comment next to dim = 2 would be sufficient.
Here, you are forcing the user to understand that it won't work in dim=1 and dim=3 by using an assertion. Why not add a comment instead that says the tutorial won't work with dim /=2 and then trust the user with it? I just think that a random user trying the tutorial would also delete line  424 and then try to move on with it, and then would be surprised that it does not work correctly

Reply:
Yes that's probably true. Nevertheless, I've prepared a commit to allow code execution for dim=1-3 (https://github.com/dealii/dealii/pull/15961/commits/95078f84da69eb8cdfccfa958f821ffe800cbb7b). This was also useful for me to see if the results produced by the code also make sense in 3D (and luckily they do :-)). Unless you are completely against it, I would prefer to keep this commit and hopefully make it less of a hassle for users :-).

Reply:
I think that's an even better solution. I am clearly not against that